prime-router/docs/proposals/0013-simplify-tables-and-value-sets.md

```diff
@@ -0,0 +1,246 @@
+It is proposed to remove the TABLE and CODE field types and use the more primitive field types like text, number, email, etc. for fields to be clearer as to what type a field really is. Table fields already require the table and tableColumn parameters, and code fields require the valueSet parameter, so those fields can be identified by the presence of those required parameters.
+
+Example table lookup with primitive field type:
+```yaml
+- name: ordering_provider_county_code
+  type: NUMBER
+  table: fips-county
+  tableColumn: FIPS
+  mapper: lookup(ordering_provider_state, ordering_provider_county)
+```
+
+### Remove the Use of Fields to Define Index Columns
+It is proposed to remove the use of fields to define the name of columns to lookup in a table for a different field. These fields that define column names have the table and tableColumn parameters, but do not use a lookup mapper. This type of field will then be changed to use a more primitive data type and will not need to specify a table or tableColumn field. This will remove the confusion as to if those index fields are looked up in a table or not, and why are they labeled as table fields if not.  The column names are to be specified in a new tableIndex parameter of the field that is doing the lookup, so the mapper no longer needs to be passed in arguments. This way, the table parameter will truly mean the table a field uses for its data, and tableColumn will truly mean the column to use for the data of the field. Furthermore, this change allows for using any column name for a given field without having to overwrite a field definition.
+
+Example of specifying lookup fields
+```yaml
+- name: ordering_provider_county_code
+  type: NUMBER
+  table: fips-county
+  tableColumn: FIPS
+  tableIndex:
+- ordering_provider_state, FIPS
+- ordering_provider_county, fips-county
+  mapper: lookup()
+```
+
+Making this change, however, results in those index fields not being populated with fake data taken from a table when generating test data.  It is proposed to change the fake data generator to look at the indices listed in each new table field and populate them with the proper data, taking care not to let the fake data generator overwrite those values as it walks through all the fields in a schema.
+
+### Support More than Two Lookup Indices
+It is proposed to support table lookups with more than two table indices.  The code is currently generic enough to support this functionality and, although we currently do not use such a feature, it will simplify the schema validation when adding the above proposed change to define the indices as a parameter.
+
+### Remove the Use of a Mapper for Table Lookups
+It is proposed to remove the use of the lookup mapper when looking up data in a table. The mapper will no longer be needed once the above proposals are implemented.
+
+Example proposed table lookup:
+```yaml
+- name: ordering_provider_county_code
+  type: NUMBER
+  pii: false
+  table: fips-county
```

Comment:
I liked the point of all "all the knowledge of the table is in one place", but what this is doing is providing all the information for the mapper. So it should be made explicit. This would also make different mappers (e.g. `livdLookup`) easy to implement, use, and read.

e.g.

```
- name: ordering_provider_county_code
  type: NUMBER
  pii: false
  mapper:
    - lookup
       table: fips-county
       tableColumn: FIPS
      etc et.c
```

Reply:
it could also be an array of mappers I suppose, leaving room for a pipeline type flow